core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -123,31 +128,46 @@ public TableScan asOfTime(long timestampMillis) {
     return useSnapshot(lastSnapshotId);
   }
 
+  @Override
+  public TableScan option(String property, String value) {
+    Map<String, String> localOptions = Maps.newHashMap();
```

Comment:
As long as this creates a new map each time, I think it makes sense to use an ImmutableMap:

```java
    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
    builder.putAll(options);
    builder.put(property, value);
    return newRefinedScan(..., builder.build());
```

Reply:
Agreed. Fixed.